simpleclient_hibernate/src/main/java/io/prometheus/client/hibernate/HibernateStatisticsCollector.java

```diff
@@ -0,0 +1,282 @@
+package io.prometheus.client.hibernate;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.CounterMetricFamily;
+import io.prometheus.client.GaugeMetricFamily;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+/**
+ * Collect metrics from Hibernate statistics.
+ * <p>
+ * Usage example:
+ * <pre>
+ * new HibernateStatisticsCollector(sessionFactory).register();
+ * </pre>
+ * If you are in a JPA environment, you can obtain the SessionFactory like this:
+ * <pre>
+ * SessionFactory sessionFactory =
+ *     entityManager.getEntityManagerFactory().unwrap(SessionFactory.class);
+ * </pre>
+ *
+ * @author Christian Kaltepoth
+ */
+public class HibernateStatisticsCollector extends Collector {
+
+  private final Statistics statistics;
+
+  public HibernateStatisticsCollector(SessionFactory sessionFactory) {
+    this(sessionFactory.getStatistics());
+  }
+
+  public HibernateStatisticsCollector(Statistics statistics) {
+    this.statistics = statistics;
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    List<MetricFamilySamples> metrics = new ArrayList<MetricFamilySamples>();
+    metrics.addAll(getSessionMetrics());
+    metrics.addAll(getConnectionMetrics());
+    metrics.addAll(getCacheMetrics());
+    metrics.addAll(getEntityMetrics());
+    metrics.addAll(getQueryExecutionMetrics());
+    return metrics;
+  }
+
+  private List<MetricFamilySamples> getSessionMetrics() {
+    return Arrays.<MetricFamilySamples>asList(
+
+        createCounter(
+            "hibernate_session_open_total",
+            "Global number of sessions opened",
```

Comment:
You are talking about the method name used to gather the metric value? Something like:

    Global number of sessions opened (getSessionOpenCount)